src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMapping.cs

```diff
@@ -64,7 +64,15 @@ public static PackageSourceMapping GetPackageSourceMapping(ISettings settings)
 
             foreach (PackageSourceMappingSourceItem packageSourceNamespaceItem in packageSourceMappingProvider.GetPackageSourceMappingItems())
             {
-                patterns.Add(packageSourceNamespaceItem.Key, new List<string>(packageSourceNamespaceItem.Patterns.Select(e => e.Pattern)));
+                if (patterns.ContainsKey(packageSourceNamespaceItem.Key))
```

Comment:
Do you need this deconstructed? 
You're adding the same KV. 

Also based on the PR title, it sounds like there shouldn't be any changes in here since this comes after.

Reply:
You're right, this code is no longer needed